libpod/container_api.go

```diff
@@ -346,7 +374,16 @@ func (c *Container) HTTPAttach(r *http.Request, w http.ResponseWriter, streams *
 	logrus.Infof("Performing HTTP Hijack attach to container %s", c.ID())
 
 	c.newContainerEvent(events.Attach)
-	return c.ociRuntime.HTTPAttach(c, r, w, streams, detachKeys, cancel, hijackDone, streamAttach, streamLogs)
+
+	err := c.ociRuntime.HTTPAttach(c, r, w, streams, detachKeys, cancel, hijackDone, streamAttach, streamLogs)
+	if err != nil {
+		if e := saveContainerError(c, err); e != nil {
```

Comment:
I think we might just want to do this on Start and StartAndAttach, but I'm not 100% sure on that, would need to verify with Docker - my understanding being that this captures errors caused by starting the container?

Reply:
That would nice since the code paths/location would be limited.

Reply:
I just did a quick look at Docker and it looks like the error message holds the last known error for start, stop, or remove https://github.com/moby/moby/blob/0ec426a57b9bcdd6cbd6fe98f7e15ebfc12d0e14/container/state.go#L32